src/resources/asres.rs

```diff
@@ -306,14 +337,54 @@ impl FromStr for AsBlocks {
 
         for el in s.split(',') {
             let el = el.trim();
-            let block = AsBlock::from_str(&el)?;
-            builder.push(block);
+            if !el.is_empty() {
+                let block = AsBlock::from_str(&el)?;
+                builder.push(block);
+            }
         }
 
         Ok(builder.finalize())
     }
 }
 
+//--- PartialEq
```

Comment:
If you implement `PartialEq` and `Eq` for `resources::chain::SharedChain` (and all the other chains, while you are at it), you can simply derive them here.

I suspect that you can even just derive them for the chains, given that they are ordered and non-continuous (i.e., the same set of resources should result in the exact same chain).

Reply:
pushed.. implemented for chains - did not try deriving there yet, Block does not implement Eq, although its Item does. Not sure.. but no time right now - have to be in a call.